test/e2e/tests/test_helper.py

```
@@ -684,6 +684,28 @@ def completions(prompt: str, model_v1: str, headers: dict, model_name: str):
     return requests.post(url, headers=headers, json=body, timeout=30, verify=TLS_VERIFY)
 
 
+# ---------------------------------------------------------------------------
+# IPP (Payload Processing) Helpers
+# ---------------------------------------------------------------------------
+
+def _check_ipp_pods_deployed():
+    """Check if payload-processing (IPP) pods are deployed in the gateway namespace."""
+    for name in ("payload-pre-processing", "payload-processing"):
+        result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)